purpleair-aqi.js

```diff
@@ -1,32 +1,71 @@
-// This code is from <https://github.com/jasonsnell/PurpleAir-AQI-Scriptable-Widget>
-// By Jason Snell based on code by Matt Silverlock, Rob Silverii, Adam Lickel
+"use strict";
 
-const API_URL = "https://www.purpleair.com/json?show=";
+/**
+ * This code is from <https://github.com/jasonsnell/PurpleAir-AQI-Scriptable-Widget>
+ * By Jason Snell based on code by Matt Silverlock, Rob Silverii, Adam Lickel, Alexander Ogilvie
+ */
 
-// Find a nearby PurpleAir sensor ID via https://fire.airnow.gov/
-// Click a sensor near your location: the ID is the trailing integers
-// https://www.purpleair.com/json has all sensors by location & ID.
-let SENSOR_ID = args.widgetParameter || "69223";
+const API_URL = "https://www.purpleair.com/json?show=";
 
-// Fetch content from PurpleAir
+/**
+ * Find a nearby PurpleAir sensor ID via https://fire.airnow.gov/
+ * Click a sensor near your location: the ID is the trailing integers
+ * https://www.purpleair.com/json has all sensors by location & ID.
+ * @type {string}
+ */
+const SENSOR_ID = args.widgetParameter || "69223";
+
+/**
+ * Widget attributes: AQI level threshold, text label, gradient start and end colors, text color
+ *
+ * @typedef {object} LevelAttribute
+ * @property {number} threshold
+ * @property {string} label
+ * @property {string} startColor
+ * @property {string} endColor
+ * @property {string} textColor
+ * @property {string} darkStartColor
+ * @property {string} darkEndColor
+ * @property {string} darkTextColor
+ */
```

Comment:
This style of writing comments is called `jsdoc`. By adding typedefs like this, we can spot places where we're converting types that don't need to be converted more easily.